src/main/java/com/ft/methodecontentcollectionmapper/client/DocumentStoreApiClient.java

```diff
@@ -90,8 +98,29 @@ public String resolveUUID(final String identifierAuthority, final String identif
 
             return uuid;
         });
-    }
-
+	}
+	
+	public boolean canResolveUUID(String uuid, String transactionId) {
+		Matcher matcher = UUID_PATTERN.matcher(uuid);
+		if (!matcher.matches()) {
+			throw new IllegalArgumentException(String.format("The uuid: %s is not valid.", uuid));
+		}
+
+		final URI contentUri = UriBuilder.fromUri(contentEndpoint).path(uuid).build();
+		LOG.info("Calling Content endpoint: {}", contentUri);
+
+		WebResource webResource = jerseyClient.resource(contentUri);
+		WebResource.Builder builder = webResource.getRequestBuilder();
+		builder = builder.header(TRANSACTION_ID_HEADER, transactionId);
+
+		final ClientResponse response = builder.get(ClientResponse.class);
+		if (response.getStatus() != HttpStatus.SC_OK) {
```

Comment:
You should return false when document-store-api returns 404.

Reply:
and no more exception thrown?

Reply:
The exception should be thrown for non 200 or 404 status codes.